hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHAWithData.java

```diff
@@ -107,7 +107,6 @@ public void testMultipartUpload() throws Exception {
   }
 
   @Test
-  @Flaky("HDDS-8035")
   public void testOMHAMetrics() throws InterruptedException,
       TimeoutException, IOException {
     waitForLeaderToBeReady();
```

Comment:
Should also fix `waitForLeaderToBeReady()`.  Currently unnecessarily checking `isLeaderReady() `within the method.  
Specifically, the line 
https://github.com/apache/ozone/blob/ce418581a8b4533a47de741d543072903562a4c6/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHAWithData.java#L181
`isLeaderReady() `is redundant as it is called within the `getCluster().getOMLeader()`
https://github.com/apache/ozone/blob/ce418581a8b4533a47de741d543072903562a4c6/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java#L197

Reply:
@neils-dev I agree and apart from that, I've also verified that during OM restart, there is no leader change taking place.

Instead of restarting the leader, we should call `getCluster().restartOzoneManager();` which restarts all OMs and triggers a new leader election.